Makefile

```diff
@@ -0,0 +1,245 @@
+
+##############################################################################
+# Variables used for various build targets.
+##############################################################################
+
+# Ensure the Go version in .go_version is installed and used.
+GOROOT?=$(shell ./scripts/make/run_with_go_ver go env GOROOT)
+GO:=$(GOROOT)/bin/go
+export PATH:=$(GOROOT)/bin:$(PATH)
+
+# By default we run tests with verbose output. This may be overridden, e.g.
+# scripts may set GOTESTFLAGS=-json to format test output for processing.
+GOTESTFLAGS?=-v
+
+GOOSBUILD:=./build/$(shell $(GO) env GOOS)
+APPROVALS=$(GOOSBUILD)/approvals
+GENPACKAGE=$(GOOSBUILD)/genpackage
+GOIMPORTS=$(GOOSBUILD)/goimports
+GOLICENSER=$(GOOSBUILD)/go-licenser
+GOLINT=$(GOOSBUILD)/golint
+MAGE=$(GOOSBUILD)/mage
+REVIEWDOG=$(GOOSBUILD)/reviewdog
+STATICCHECK=$(GOOSBUILD)/staticcheck
+ELASTICPACKAGE=$(GOOSBUILD)/elastic-package
+
+PYTHON_ENV?=.
+PYTHON_BIN:=$(PYTHON_ENV)/build/ve/$(shell $(GO) env GOOS)/bin
+PYTHON=$(PYTHON_BIN)/python
+
+CLOUDBEAT_VERSION=$(shell grep defaultBeatVersion cmd/version.go | cut -d'=' -f2 | tr -d '" ')
+
+# Create a local config.mk file to override configuration,
+# e.g. for setting "GOLINT_UPSTREAM".
+-include config.mk
+
+##############################################################################
+# Rules for building and unit-testing cloudbeat.
+##############################################################################
+
+.DEFAULT_GOAL := cloudbeat
+
+.PHONY: cloudbeat
+cloudbeat:
+	@$(GO) build -o $@
+
+.PHONY: test
+test:
+	$(GO) test $(GOTESTFLAGS) ./...
+
+.PHONY:
+clean: $(MAGE)
+	@$(MAGE) clean
+
+##############################################################################
+# Checks/tests.
+##############################################################################
+
+.PHONY: check-full
+# check-full: update check golint staticcheck check-docker-compose
+check-full: update
+
+.PHONY: check-approvals
+check-approvals: $(APPROVALS)
+	@$(APPROVALS)
+
+.PHONY: check
+check: $(MAGE) check-fmt check-headers
+	@$(MAGE) check
+
+.PHONY: bench
+bench:
+	@$(GO) test -benchmem -run=XXX -benchtime=100ms -bench='.*' ./...
+
+##############################################################################
+# Rules for updating config files, etc.
+##############################################################################
+
+#update: go-generate add-headers build-package notice $(MAGE)
+update: go-generate add-headers $(MAGE)
+	@$(MAGE) update
+	@go mod download all # make sure go.sum is complete
+
+config:
+	@$(MAGE) config
+
+.PHONY: go-generate
+go-generate:
+	@$(GO) generate .
+
+notice: NOTICE.txt
+NOTICE.txt: $(PYTHON) go.mod utils/go.mod
+	@$(PYTHON) scripts/make/generate_notice.py . 
+
+.PHONY: add-headers
+add-headers: $(GOLICENSER)
+ifndef CHECK_HEADERS_DISABLED
+	@$(GOLICENSER)
+#	@$(GOLICENSER) -license Elasticv2 x-pack
```

Comment:
can be removed, no x-pack here or planned to be afaik

Reply:
removed